### ISSUE TITLE
Application name based Geometry

### ISSUE THREAD

**Issue by coxley** [knopwob/dunst#265]
Hi all,

I have a use-case that may not be super common but would be very helpful for me. I have some applications that I would like to use different geometry than the global `geometry` variable (e.g, `0x5-30+20` so these applications aren't width-limited) but others that I want to follow the global value.

Reason for this is some application notifications have a specific format that only looks right with a monospaced font and width is unpredictable. Like lining up columns.

Is this something possible or easy to add?

**Comment by neg-serg** [comment]
I am really want this feature too!
Consider that we have mpd-notification like here:
https://i.imgur.com/MZsbKOR.png

It would be very glad to see another notifications in the different area.

**Comment by quite** [comment]
I agree that we must be able to set geometry dynamically. I'd say what is needed is ability to use `geometry` in category sections.

**Comment by Lundmore** [comment]
I too have a use-case for this feature. It would make dunst much more dynamic!

**Comment by Flurrywinde** [comment]
It seems like this feature isn't implemented yet? Too bad! I too have a use-case.

Today, I decided to experiment, and I discovered one can do something like: `xdotool search --class Dunst windowmove 0 0` to move it.

However, the next time it opens, it'll be in that place again.

However, you can move it back to its default location later (like after calling `sleep`), and it doesn't have to be visible at the time.

I hope this helps some people (and doesn't have side-effects I haven't discovered yet).

**Comment by s0la** [comment]
Any progress on this?

I use dunst quite heavily and I could greatly benefit from such a feature, given I use it with a lot of different custom applications, with a different placement, dimensions and format.. Right now, I basically restart dunst whenever I need a different configuration, but I have one usecase where it acts as a "visual representation" for windows resizing using keyboard, and killing and starting introduces a bit of latency, so the whole process doesn't seem that smooth..

I'd be really interested in this feature as well! :)

**Comment by bynect** [comment]
> Any progress on this?
> 
> I use dunst quite heavily and I could greatly benefit from such a feature, given I use it with a lot of different custom applications, with a different placement, dimensions and format.. Right now, I basically restart dunst whenever I need a different configuration, but I have one usecase where it acts as a "visual representation" for windows resizing using keyboard, and killing and starting introduces a bit of latency, so the whole process doesn't seem that smooth..
> 
> I'd be really interested in this feature as well! :)
> 
> EDIT:
> 
> Btw, just out of curiosity, what is the difference between changing let's say background color and geometry through custom rules? Isn't that just another variable that has to temporarily replace its original value by applying the specific rules?

the fact is that at the moment dunst uses a single window for all notifications. So you can change the layout and color of the notification, but changing their position is trickier. Because you have to change the whole window position.
